### PR TITLE
add claude35haiku & not support vision

### PR DESCRIPTION
**Pull request by OPChips**
#### 💻 变更类型 | Change Type

- [ ] feat    
- [ ] fix    
- [ ] refactor    
- [ ] perf    
- [ ] style    
- [ ] test    
- [ ] docs    
- [ ] ci    
- [ ] chore    
- [ ] build    

## Summary by CodeRabbit

- **New Features**
	- Added the model identifiers `"claude-3-5-haiku-20241022"` and `"claude-3-5-haiku-latest"` to enhance model options for the Anthropic service.

- **Bug Fixes**
	- Updated logic in the model evaluation process to exclude the new model from certain evaluations.

- **Chores**
	- Improved the `fetch` function to utilize `tauriStreamFetch` for better compatibility.

### CODE REVIEW

**Comment by lloydzhou** [comment]
有合最新的代码吗？

File: app/constant.ts [hunk 1]
```diff
@@ -329,6 +329,7 @@ const anthropicModels = [
   "claude-3-opus-20240229",
   "claude-3-opus-latest",
   "claude-3-haiku-20240307",
+  "claude-3-5-haiku-20241022",
```
Comment (lloydzhou) [inline app/constant.ts:332]: 排序，把claude-3-5-haiku-latest和claude-3-5-haiku-20241022放一起

File: app/utils.ts [hunk 2]
```diff
@@ -254,6 +254,7 @@ export function getMessageImages(message: RequestMessage): string[] {
 export function isVisionModel(model: string) {
   // Note: This is a better way using the TypeScript feature instead of `&&` or `||` (ts v5.5.0-dev.20240314 I've been using)
 
+  const excludeKeywords = ["claude-3-5-haiku-20241022"];
```
Comment (lloydzhou) [inline app/utils.ts:257]: 这里应该先判断属于排除列表直接返回

现在放后面通过`&&`比较，会多出来一些多余的判断逻辑。

**Review by lloydzhou — APPROVED** [review]

File: app/utils.ts (hunk 2 above)
Comment (OPChips) [reply app/utils.ts:257]: 这里&&短路了 没问题

Comment (Dogtiti) [reply app/utils.ts:257]: 意思是excludeKeywords判断直接return 可以避免后面的运算
当然&&也没问题

**Review by Dogtiti — DISMISSED** [review]

**Pull request merged**